wellcometrust/design-system
# 8709 task: design tokens used to output styles and documentation (colours)

Relates to wellcometrust/corporate#8709

## Context
Design tokens will be the core of the design system long-term. Initially however the aim is to automate the process of outputting CSS variables for styling and also documentation for Storybook from a single source (i.e. a JSON file `src/tokens/colour/base.json`). Amazon's Style Dictionary enables us to do this very easily and for the long-term could be used to output other style formats e.g. iOS, Android, Kotlin.

This PR is a proof of concept for flat variables. Further discovery needed to work out how to output CSS containing media queries

Design token build files (`src/build/css/colours.css`, `src/build/css/colours.js` and `src/build/css/colours.d.ts`) are included in this commit and not specifically ignored as they are dependencies for the CSS and Storybook output.

## To test
To run locally:

1. pull this branch
2. run `npm install`
3. run `npm run storybook` and check Globals > Colours
4. optionally, to preview how design tokens are generated run `npm run build:tokens` to output css, js and d.ts to `src/build`
5. run `npm run build` to output React and CSS for consumption in the `dist` folder
